### PR TITLE
Fully specify all fields within `signatures`

### LINKED ISSUE #4: Specify the other fields of `signature`
The current spec lists the other fields as `...` because it was originally written as a diff from in-toto's current format ([ITE-5](https://github.com/in-toto/ITE/pull/13)). This needs to be fleshed out for the spec to be stand-alone.

There is only one field to define: `keyid`. The meaning and semantics of this field are currently under discussion. @SantiagoTorres, @TomHennen, @nenaddedic, and I talked about this in our last meeting, but we didn't come to any conclusion. In-toto [currently](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#42-file-formats-general-principles) defines KEYID as a cryptographic hash over some representation of the key. However, it may be valuable to leave this user-defined in case users want to use an identifier with their key management system of choice.

### PR DESCRIPTION
**Pull request by MarkLodato**
Closes #4.

### CODE REVIEW

File: specification.md [hunk 1]
```diff
@@ -54,6 +57,11 @@ where:
     -   https://theupdateframework.com/Root/v1.0.5
     -   etc...
 
+*   KEYID is an optional, unauthenticated hint indicating what key was used to
+    sign the message. It **must not** be used for security decisions.
```
Comment (joshuagl) [inline specification.md:61]:
```suggestion
    sign the message. It **MUST NOT** be used for security decisions.
```
In TUF we uppercase terms from [RFC 2119](https://tools.ietf.org/html/rfc2119) and refer readers there for interpretation guidelines.

File: specification.md [hunk 2]
```diff
@@ -54,6 +57,11 @@ where:
     -   https://theupdateframework.com/Root/v1.0.5
     -   etc...
 
+*   KEYID is an optional, unauthenticated hint indicating what key was used to
```
Comment (joshuagl) [inline specification.md:60]: While I realise that signing-spec does not currently define `Sign()` or talk about how to use KEYID...

Is KEYID currently optional in in-toto? Do we anticipate that the KEYID is only omitted in cases where there is only one valid key? Can we provide some additional guidance here?

File: specification.md [hunk 3]
```diff
@@ -121,6 +130,7 @@ To sign:
 -   BODY **must** be an object type (`{...}`).
 -   Serialize BODY as [Canonical JSON]; call this SERIALIZED_BODY.
 -   Sign SERIALIZED_BODY, base64-encode the result, and store it in `sig`.
+-   Optionally, compute a KEYID and store it in `keyid`.
```
Comment (joshuagl) [inline specification.md:133]: Should we be discussing/describing KEYID computation? Is that a future enhancement?

File: specification.md (hunk 1 above)
Comment (MarkLodato) [reply specification.md]: Thanks! Done.

File: specification.md (hunk 3 above)
Comment (MarkLodato) [reply specification.md:133]: I suggest leaving that to #9.

Comment (joshuagl) [reply specification.md:136]: Sounds reasonable, I appreciate the change you have made to expand on KEYID.

File: specification.md (hunk 2 above)
Comment (MarkLodato) [reply specification.md]: I actually don't know about the in-toto side. Perhaps some in-toto expert can answer? Also, please note that full specification of KEYID and Sign() is covered by issue #9.

Please take a look at the most recent version. I expanded the text to hopefully better explain:
- It is negotiated out of band.
- It is only used to narrow the selection of keys.

Do you think this is sufficient for the time being?

File: specification.md (hunk 3 above)
Comment (MarkLodato) [reply specification.md:136]: Wow, you're a fast reviewer! :-D

(Marking resolved.)

File: specification.md (hunk 2 above)
Comment (joshuagl) [reply specification.md]: I do, this looks good thanks Mark!

File: specification.md [hunk 4]
```diff
@@ -134,16 +147,35 @@ To verify:
...
-    SERIALIZED_BODY must be `{`. Verifiers **must not** require SERIALIZED_BODY
+    SERIALIZED_BODY **MUST** be `{`. Verifiers **MUST NOT** require SERIALIZED_BODY
     to be Canonical JSON.
 
 Backwards compatible signatures are not recommended because they lack the
 authenticated payloadType indicator.
 
 This scheme is safe from rollback attacks because the first byte of
-SERIALIZED_BODY must be 0x7b (`{`) in backwards compatibility mode and 0x02 in
+SERIALIZED_BODY is be 0x7b (`{`) in backwards compatibility mode and 0x02 in
```
Comment (joshuagl) [inline specification.md:157]:
```suggestion
SERIALIZED_BODY is 0x7b (`{`) in backwards compatibility mode and 0x02 in
```

**Review by joshuagl — APPROVED** [review]
Minor grammatical nit, but otherwise this looks great. Thanks!

**Review by adityasaky — APPROVED** [review]
Thanks, @MarkLodato!

**Pull request merged**